candidates/tests/test_mark_change_as_reviewed_view.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import User
+from django_webtest import WebTest
+
+from .auth import TestUserMixin
+from ..models import ChangeReviewed, LoggedAction
+from candidates.tests import factories
+
+from popolo.models import Person
+
+
+class TestMarkChangeAsReviewedView(TestUserMixin, WebTest):
+    def setUp(self):
+        test_person_1 = factories.PersonExtraFactory.create(
+            base__id=9876,
+            base__name='Test Candidate for Recent Changes',
+        )
+        self.action1 = LoggedAction.objects.create(
+            user=self.user,
+            action_type='person-create',
+            ip_address='127.0.0.1',
+            person=test_person_1.base,
+            popit_person_new_version='1234567890abcdef',
+            source='Just for tests...',
+        )
+        self.app.get(
+            '/recent-changes',
+            user=self.user,
+        )
+
+    def tearDown(self):
+        self.action1.delete()
+
+    def test_redirects_to_recent_changes_page(self):
+        response = self.post_to_change_reviewed(
+            Person.objects.last().id,
+            LoggedAction.objects.last().id,
+            self.user.id
+        )
+        self.assertEqual(response.status_code, 302)
+
+    def test_marks_a_change_as_reviewed(self):
+        count_before = len(ChangeReviewed.objects.all())
+        self.post_to_change_reviewed(
+            Person.objects.last().id,
+            LoggedAction.objects.last().id,
+            self.user.id
+        )
+        count_after = len(ChangeReviewed.objects.all())
+        self.assertEqual(count_after - count_before, 1)
+
+    def test_errors_if_form_is_invalid(self):
+        person_id = Person.objects.last().id
+        logged_action_id = LoggedAction.objects.last().id
+        reviewer_id = self.user.id
+
+        with self.assertRaises(Exception):
+            self.post_to_change_reviewed(None, logged_action_id, reviewer_id)
+
+        with self.assertRaises(Exception):
+            self.post_to_change_reviewed(person_id, None, reviewer_id)
+
+        with self.assertRaises(Exception):
+            self.post_to_change_reviewed(person_id, logged_action_id, None)
```

Comment:
I'd probably make these three separate test methods (`test_error_on_bad_person_id`, `test_error_on_bad_logged_action_id`, etc.) so that if you get a test failure it's more descriptive and you avoid the "fix one test failure, oh, that was also hiding a subsequent test failure" loop...

Also, asserting that something raises `Exception` isn't as helpful as it could be - that might include an error from mistyping a variable name, say, and not the exception you're expecting to be raised. I think you should catch a more specific exception class.